lib/reverse_sentence.rb

```diff
@@ -1,6 +1,136 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(2n) The array is reviewed twice, once for spaces to populate the space array, and once more piecewise to do all the swaps
+# Space complexity: O(1)
```

Comment:
You are building a space array, so this would actually be:  O(m) where m is the number of words.